# Time Value of Money 

A dollar today is worth more than a dollar tomorrow. This simple observation underpins vitually every financial calculation you will encounter, from pricing bonds and valuing companies to structuring mortgages and retirment plans. The concept is called **Time Value of Money**(TVM) and understanding it is essential for anyone working in quantitative finance.


Why is money worth more today? Three reasons drive this phenomenon
1. The money received today can be invested to earn a return, making it grow over time.
2. Inflation erodes purchasing power, so a dollar today buys more than a dollar in the future.
3. There is uncertainty about whether promised future payments will materialize.

These factors combine to create a fundamental asymmetry: present money is more valuable than future money.

The time value of money gives us a framework for comparing cash flows that occur at different points in time. Without this framework, how would you choose between receiving $1,000 today versus $1,100 in one year? Or evaluate a business that requires $1 million upfront but promises $200,000 annually for the next seven years? TVM provides the mathematical machinery to answer these questions rigorously.

## 1. Future Value and Compounding:

When  you deposit money in an interest-bearing account, you balance grows over time. The process by which this growth occurs and particularly how interest accu,ulates on previously earned interest is called **compounding**. This concept drives both wealth accumulation and debt growth. Lets build up from the simplest case to the continous limit used in derivative pricing understanding each step along the way.

### a. Simple Interest

Simple interest is the most basic form of interest calculation and is our conceptual starting point. Under simple interest, the interest accrues only on the original principal amount, never on accumulated interest. Think of it as a linear process : each period adds the same fixed amount of interest, regardless of how much has accumulated before. If you invest a principal 
$P$ at an annual interest rate 
$r$ for $t$ years, the future value is:

$$FV = P (1 + r * t)$$

where :

-  $FV$ : Future Value of the investment
-  $P$: Principla amount(intial investment)
-  $r$: annual interest rate(expressed as a decimal)
-  $t$: time in years

The structure of this formula reveals its linear nature. The term $rt$ represents the total proportional increase: rate $r$ multiplied by time $t$. When you add this to $1$(representing the original principal as `100%`), you get the total growth factors. Multiplying by $P$ then scales this to your actual investment amount. Notice that time and rate enter symmetrically and additively. Doubling the time has exactly the same effect as doubling the rate.

>Simple interest is calculated only on the original principal amount. The interest earned each period remains constant regardless of how long the investment is held.


For example, investing `$1,000` at `5%` simple interest for `3` years yields:

In [1]:
P = 1000 # Principal
r = 0.05 # Annual Interest rate
t = 3 # Time in years


Fv_Simple = P * ( 1 + r * t)
interest_earned = Fv_Simple - P

print(f"Principal:{P}")
print(f"Future Value:{Fv_Simple}")
print(f"Interest earned:{interest_earned}")


Principal:1000
Future Value:1150.0
Interest earned:150.0


In [2]:
# Simple Interest Function
def SimpleInterest(P,r,t):
    Fv_Simple = P * ( 1 + r * t)
    interest_earned = Fv_Simple - P
    return Fv_Simple, interest_earned



Notice that the interest earned ($150) equals the annual interest ($50) multiplied by the number of years (3). Simple interest grows linearly with time, which makes it straightforward to calculate but unrealistic for most financial applications. In the real world, lenders and investors expect to earn interest on their accumulated interest, a concept we turn to next.